ipi/engine/motion/dynamics.py

```diff
@@ -118,6 +125,17 @@ def __init__(
         else:
             self.fixatoms = fixatoms
 
+        self.eda_on = False  # whether the dynamics is driven or not
```

Comment:
I think there's a lot of unnecessary redundancy in this implementation. It should be possible to implement this as nvtcc - just a change in integrator. Alternatively, if this requires heavier modifications of the MD loop, it'd be best to implement it taking as example the Planetary class. Oveall, I feel it'd be much better to treat this as a different Motion type like planetary than as a further mode of Dynamics, that is already very complex with MTS and all.

Reply:
I created a `DrivenDynamics` class, which inherits from `Dynamics`, and can be specified with the keyword `driven_dynamics` in the `input.xml` file (have a look at the examples).
All the rest of my code has not been modified substantially.

Reply:
My "old" classes are still in `ipi/engine/motion/eda.py` and `ipi/input/motion/eda.py`.
I created two new files, `ipi/engine/motion/driven_dynamics.py` and `ipi/input/motion/driven_dynamics.py`, to properly implement a new `Motion` class that inherits from `Dynamics`.
I can move all the contents of the `eda.py` files into `driven_dynamics.py`, but if anybody else is interested in implementing something similar, it could be better to have a separate class for `driven dynamics` and then a different file for all possible the ways/flavors to do that (`eda.py` in my case, `some-other-driven-dynamics.py` for somebody else).

Reply:
I agree that we could see many types of ("time")-driven dynamics.